src/__tests__/validatePath.test.ts

```diff
@@ -132,6 +132,38 @@ describe("unit tests for the parser", () => {
       path: "States.MathAdd($.value1, $.step)",
       valid_in: PayloadTemplatesOnly,
     },
+    {
+      path: "States.MathAdd($.value1, -1)",
+      valid_in: PayloadTemplatesOnly,
+    },
+    {
+      path: "States.MathAdd($.value1, 1)",
+      valid_in: PayloadTemplatesOnly,
+    },
+    {
+      path: "States.MathAdd(1, -1)",
+      valid_in: PayloadTemplatesOnly,
+    },
+    {
+      path: "States.MathAdd(-1, -1)",
+      valid_in: PayloadTemplatesOnly,
+    },
+    {
+      path: "States.MathAdd(1, $.step)",
+      valid_in: PayloadTemplatesOnly,
+    },
+    {
+      path: "States.MathAdd(-1, $.step)",
+      valid_in: PayloadTemplatesOnly,
+    },
+    {
+      path: "States.MathAdd('-1', 'abc')",
+      valid_in: None,
+    },
+    {
+      path: "States.MathAdd(States.ArrayLength($$.Execution.Input.machines),-1)",
```

Comment:
this is the expression from the original bug report here: https://github.com/ChristopheBougere/asl-validator/issues/165